lib/ui/setup/pair_page.dart

```diff
@@ -1,234 +1,188 @@
+      addressWrapper.value = dev.address;
+      uiConnectionControl.connectToWatch(addressWrapper);
+    };
+
+    return Scaffold(
+      appBar: AppBar(
+        title: Text("Pair a watch"),
+        leading: BackButton(),
+      ),
+      body: ListView(
+        children: [
+          if (scan.scanning)
+            Padding(
+              padding: EdgeInsets.all(16.0),
+              child: UnconstrainedBox(
+                child: CircularProgressIndicator(),
+              ),
+            ),
+          ...scan.devices
+              .map(
+                (e) => InkWell(
+                  child: Container(
+                    child: Row(
+                      children: <Widget>[
+                        Container(
+                          child: Center(
+                            child: PebbleWatchIcon(
+                              PebbleWatchModel.values[e.color],
+                            ),
                           ),
-                          FlatButton(
-                            child: Text("SEARCH AGAIN WITH BT CLASSIC"),
-                            padding: EdgeInsets.symmetric(horizontal: 32.0),
-                            textColor: Theme.of(context).accentColor,
-                            onPressed: _refreshDevicesClassic,
-                          )
-                        ],
-                      ),
-                    ),
-                    if (widget.showSkipButton)
-                      FlatButton(
-                        child: Text("SKIP"),
-                        padding: EdgeInsets.symmetric(horizontal: 32.0),
-                        onPressed: () => context.pushAndRemoveAllBelow(
-                          HomePage(),
+                          width: 56,
+                          height: 56,
+                          decoration: BoxDecoration(
+                              color: Theme.of(context).dividerColor,
+                              shape: BoxShape.circle),
                         ),
-                      )
-                  ]))
-        ]));
-  }
-
-  @override
-  void dispose() {
-    super.dispose();
-    ScanCallbacks.setup(null);
-    PairCallbacks.setup(null);
+                        SizedBox(width: 16),
+                        Column(
+                          children: <Widget>[
+                            Text(
+                              e.name,
+                              style: TextStyle(fontSize: 16),
+                            ),
+                            SizedBox(height: 4),
+                            Text(
+                              e.address
+                                  .toRadixString(16)
+                                  .padLeft(6, '0')
+                                  .toUpperCase(),
+                            ),
+                            Wrap(
+                              spacing: 4,
+                              children: [
+                                Offstage(
```

Comment:
It seems you wish to simply hide widgets, right? In that case, it's better to simply skip rendering them, Offstage widget will still render them, consuming CPU and memory. You could write something like this:

```
children: [
  if(e.runningPRF && !e.firstUse)
    Chip(
      backgroundColor: Colors.deepOrange,
      label: Text("Recovery"),
    ),
]
```